runtimeconfig/manager.go

```diff
@@ -142,16 +144,30 @@ func (om *Manager) loop(ctx context.Context) error {
 	}
 }
 
-// loadConfig loads configuration using the loader function, and if successful,
-// stores it as current configuration and notifies listeners.
+// loadConfig loads all configuration files using the loader function, merges the yaml configuration files into one yaml document,
+// and notifies listeners if successful.
 func (om *Manager) loadConfig() error {
-	buf, err := os.ReadFile(om.cfg.LoadPath)
+	mergedConfig := map[string]interface{}{}
+	for _, f := range strings.Split(om.cfg.LoadPath, ",") {
+		yamlFile := map[string]interface{}{}
+		buf, err := os.ReadFile(f)
+		if err != nil {
+			om.configLoadSuccess.Set(0)
+			return errors.Wrap(err, "read file")
+		}
+		err = yaml.Unmarshal(buf, &yamlFile)
+		if err != nil {
```

Comment:
Ignorable nit, but this is more idiomatic and it we actually create a new error variable:

```suggestion
		if err := yaml.Unmarshal(buf, &yamlFile); err != nil {
```